server/src/main/java/org/opensearch/cluster/node/DiscoveryNodes.java

```diff
@@ -419,46 +419,47 @@ public DiscoveryNode resolveNode(String node) {
                             } else {
                                 resolvedNodesIds.removeAll(dataNodes.keys());
                             }
-                        } else if (DiscoveryNodeRole.MASTER_ROLE.roleName().equals(matchAttrName)) {
-                            if (Booleans.parseBoolean(matchAttrValue, true)) {
-                                resolvedNodesIds.addAll(masterNodes.keys());
-                            } else {
-                                resolvedNodesIds.removeAll(masterNodes.keys());
-                            }
-                        } else if (DiscoveryNodeRole.INGEST_ROLE.roleName().equals(matchAttrName)) {
-                            if (Booleans.parseBoolean(matchAttrValue, true)) {
-                                resolvedNodesIds.addAll(ingestNodes.keys());
-                            } else {
-                                resolvedNodesIds.removeAll(ingestNodes.keys());
-                            }
-                        } else if (DiscoveryNode.COORDINATING_ONLY.equals(matchAttrName)) {
-                            if (Booleans.parseBoolean(matchAttrValue, true)) {
-                                resolvedNodesIds.addAll(getCoordinatingOnlyNodes().keys());
+                        } else if (DiscoveryNodeRole.MASTER_ROLE.roleName().equals(matchAttrName)
+                            || DiscoveryNodeRole.CLUSTER_MANAGER_ROLE.roleName().equals(matchAttrName)) {
```

Comment:
Note: Most of the changes around are indent change, and this line is the actual change:
`|| DiscoveryNodeRole.CLUSTER_MANAGER_ROLE.roleName().equals(matchAttrName)`

Reply:
Note: with @reta suggestion (https://github.com/opensearch-project/OpenSearch/pull/2424#discussion_r829126035). I wrapped up the `||` logic in a private method in this class to make the code clearer. (in the commit https://github.com/opensearch-project/OpenSearch/commit/a2661a83180507967e63f964e6b521b127bf0bff)